MetaMask/eth-snap-keyring
# fix: reject unsupported account methods

Current state: When you update an account and remove a method, the removed method still functions
Solution: Check if an account supports the method when a request is submitted to the Keyring and if not, throw an error.

* Fixes [#121](https://github.com/MetaMask/accounts-planning/issues/121)

## Manual Testing

1. Go to the [SSK](https://metamask.github.io/snap-simple-keyring/latest/) and create an account
2. Updated the account using this JSON, replacing the address and ID fields with the new account information, and choose one or more of the methods to remove from the list
```json
{
    "address": "you new account address",
    "id": "your new account ID",
    "options": {},
    "methods": [
        "personal_sign",
        "eth_sign",
        "eth_signTransaction",
        "eth_signTypedData_v1",
        "eth_signTypedData_v3",
        "eth_signTypedData_v4"
    ],
    "type": "eip155:eoa"
}
```
3. Navigate to the [E2E Test Dapp](https://metamask.github.io/test-dapp/) and connect the new account
4. Attempt to perform a method you just removed
5. You should see a message that "Method [removed method] is not supported for account [new snap account address]"

### Jest
`jest src/SnapKeyring.test.ts --testNamePattern="fails when the EthMethod is not supported"`
` jest src/SnapKeyring.test.ts`
